homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,222 @@
+            step_id="user",
+            errors=errors,
+            description_placeholders=PLACEHOLDERS,
+            data_schema=self.add_suggested_values_to_schema(CONFIG_SCHEMA, user_input),
+        )
+
+    async def async_step_friends(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Form to select other users and friends."""
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for username in user_input[CONF_USERS]:
+                lastfm_user = get_lastfm_user(self.data[CONF_API_KEY], username)
+                lastfm_errors = validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(username)
+            user_input[CONF_USERS] = valid_users
+            if not errors:
+                return self.async_create_entry(
+                    title="LastFM",
+                    data={},
+                    options={
+                        CONF_API_KEY: self.data[CONF_API_KEY],
+                        CONF_MAIN_USER: self.data[CONF_MAIN_USER],
+                        CONF_USERS: [
+                            self.data[CONF_MAIN_USER],
+                            *user_input[CONF_USERS],
+                        ],
+                    },
+                )
+        try:
+            main_user = get_lastfm_user(
+                self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+            )
+            friends = [
+                SelectOptionDict(value=friend.name, label=friend.get_name(True))
+                for friend in main_user.get_friends()
+            ]
+        except WSError:
+            friends = []
+        return self.async_show_form(
+            step_id="friends",
+            errors=errors,
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERS): SelectSelector(
+                            SelectSelectorConfig(
+                                options=friends, custom_value=True, multiple=True
+                            )
+                        ),
+                    }
+                ),
+                user_input or {CONF_USERS: []},
+            ),
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import config from yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        return self.async_create_entry(
+            title="LastFM",
+            data={},
+            options={
+                CONF_API_KEY: import_config[CONF_API_KEY],
+                CONF_MAIN_USER: None,
+                CONF_USERS: import_config[CONF_USERS],
+            },
+        )
```

Comment:
It's probably better to not create the entry here but take the `import_config` and send it to friends step to get validation on the friends list.

Reply:
Yes but what if a user doesn't exist? It would fail validation and then it would not import I guess? and now it would create an entity that would be unavailable, so people can remove that name manually from the list.

Reply:
True, but then we should probably have the validation in the import too but only save the ones that works instead of throwing an error.

Reply:
For importing that'd be my preferred method. But for options/config flow I think giving the user some feedback would be a good thing instead of not just ignoring the issue

Reply:
Agree